### PR TITLE
make it possible to passthrough functions to the swagger-initializer.js

### LINKED ISSUE #25: uiConfig cant handle functions
### Prerequisites

- [X] I have written a descriptive issue title
- [X] I have searched existing issues to ensure the issue has not already been raised

### Issue

Hello!

Firstly, thanks for an awesome package! Such a huge help. 

I got a question and I am hoping you can help me out on this. I am not quite sure but it seems that preauthorizeBasic and preauthorizeApiKey instance methods as documented in https://github.com/swagger-api/swagger-ui/blob/master/docs/usage/configuration.md are not yet supported by the package. 

Are these options something that you plan to support in the future? If I overlooked it and that they're already supported, can you please provide an example on how I can pass them?

In swagger-ui-express, I am able to utilize the onComplete parameter to pass a function that invokes those instance methods as shown below:
 {
           tagsSorter: 'alpha',
            **onComplete: function () {
                    ui.preauthorizeBasic("basicAuth", "user", "pass");
                    ui.preauthorizeApiKey("apiKey", "apikey");
            }**   
}

Thanks for the feedback in advance!

### PR DESCRIPTION
**Pull request by Uzlopak**
Closes #25

I know that the coverage is failing, but writing 100% test coverage would be a waste of time if the approach would not be accepted.

So currently we JSON.stringify uiConfig and initOAuth which is basically removing functions as functions are not per definition serializable. But with my super simple serializiation function, I can serialize an Object as best as possible. Yeah sure... ES6 Maps and Sets and what not are not supported or yet. But whatever. Supplying the initOAuth and uiConfig per swagger-initialization.js so we can support functions.

Removes the initOAuth and the uiConfig routes as they dont need to be fetched again.

#### Checklist

- [ ] run `npm run test` and `npm run benchmark`
- [ ] tests and/or benchmarks are included
- [ ] documentation is changed or added
- [ ] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

File: lib/routes.js [hunk 1]
```diff
@@ -74,23 +75,55 @@ function fastifySwagger (fastify, opts, done) {
...
   fastify.route({
-    url: '/initOAuth',
+    url: '/static/swagger-initializer.js',
     method: 'GET',
     schema: { hide: true },
     ...hooks,
     handler: (req, reply) => {
-      reply.send(opts.initOAuth)
+      reply
+        .header('content-type', 'application/javascript; charset=UTF-8')
```
Comment (Fdawgs) [inline lib/routes.js:125]:
```suggestion
        .header('content-type', 'application/javascript; charset=utf-8')
```

I know charset is case insensitive, but maybe make this lowercase to keep in line with what Fastify uses?

**Comment by Fdawgs** [comment]
Docs would need to be updated to mention no support for Maps, Sets etc. until they're added, right?

File: test/route.test.js [hunk 2]
```diff
@@ -361,13 +305,6 @@ test('/documentation/static/:file should send back the correct file', async (t)
     })
     t.equal(typeof res.payload, 'string')
     t.equal(res.headers['content-type'], 'application/javascript; charset=UTF-8')
```
Comment (Uzlopak) [inline test/route.test.js:307]:
```suggestion
    t.equal(res.headers['content-type'], 'application/javascript; charset=utf-8')
```

**Comment by Uzlopak** [comment]
Ok, seems like you have interest in this solution. Added Maps and Sets. but tbh... they werend supported till now neither. 

 I think the solution is complete.

File: README.md [hunk 3]
```diff
@@ -108,7 +108,7 @@ await fastify.ready()
  | transformStaticCSP   | undefined        | Synchronous function to transform CSP header for static resources if the header has been previously set.                  |
  | transformSpecification     | undefined        | Synchronous function to transform the swagger document.                                                                   |
  | transformSpecificationClone| true             | Provide a deepcloned swaggerObject to transformSpecification                                                                    |
- | uiConfig             | {}               | Configuration options for [Swagger UI](https://github.com/swagger-api/swagger-ui/blob/master/docs/usage/configuration.md). Must be literal values, see [#5710](https://github.com/swagger-api/swagger-ui/issues/5710).                                                                                                            |
+ | uiConfig             | {}               | Configuration options for [Swagger UI](https://github.com/swagger-api/swagger-ui/blob/master/docs/usage/configuration.md).                                                                                                           |
```
Comment (climba03003) [inline README.md:111]: Please add a detail on `function` should be self contained which means any variables outside the `function` would not works.

**Review by climba03003 — CHANGES_REQUESTED** [review]

File: README.md (hunk 3 above)
Comment (Uzlopak) [reply README.md:111]: Actually the swagger UI element is bound to window.ui. so you could access that from the function.

```suggestion
 | uiConfig             | {}               | Configuration options for [Swagger UI](https://github.com/swagger-api/swagger-ui/blob/master/docs/usage/configuration.md). Functions should be self contained, as variables outside the function are potentially not working. You can access the swagger-ui element with `ui` as it is assigned to `window.ui`.                                                                                                   |
```

Like th8s?

Comment (climba03003) [reply README.md:111]: > Like this?

`potentially not working` is would give a strong stance here `not working`.

Comment (Uzlopak) [reply README.md:111]:
```suggestion
 | uiConfig             | {}               | Configuration options for [Swagger UI](https://github.com/swagger-api/swagger-ui/blob/master/docs/usage/configuration.md). Functions should be self contained, as variables outside the functions are  not working. You can access the swagger-ui element with `ui` as it is assigned to `window.ui`.                                                                                                   |
```

Comment (Uzlopak) [reply README.md:111]: Well I had a good laugh about the example. But what about accessing global scope window? :D I was thinking of that and not thinking to access fastify backend vaiables :D

Comment (Uzlopak) [reply README.md]: Adapted it

**Review by Fdawgs — APPROVED** [review]
LGTM, as Climba said, need some more eyes on it.

Is the CI linting failure an issue?

**Comment by climba03003** [comment]
> Is the CI linting failure an issue?

Seems like a bug of `standard`.
It should not crash itself as a linting tools.

**Comment by Uzlopak** [comment]
I investigated why suddenly linting is failing, and I have no clue. Locally it works. If I remove the standard rules in package.json i get the same error as in the ci.

**Comment by Uzlopak** [comment]
The linting issue can be found here https://github.com/estools/esquery/issues/135
Not related to our code

**Comment by Uzlopak** [comment]
Ok. Linting is working now after equery was fixed

File: lib/routes.js [hunk 4]
```diff
@@ -74,23 +75,17 @@ function fastifySwagger (fastify, opts, done) {
...
-    ...hooks,
-    handler: (req, reply) => {
-      reply.send(opts.uiConfig)
-    }
-  })
+  const swaggerInitializerContent = swaggerInitializer(opts)
 
   fastify.route({
-    url: '/initOAuth',
+    url: '/static/swagger-initializer.js',
```
Comment (Eomm) [inline lib/routes.js:81]:
```suggestion
    url: `${staticPrefix}/swagger-initializer.js`,
```

File: README.md [hunk 5]
```diff
@@ -108,7 +108,7 @@ await fastify.ready()
...
  | transformSpecification     | undefined        | Synchronous function to transform the swagger document.                                                                   |
  | transformSpecificationClone| true             | Provide a deepcloned swaggerObject to transformSpecification                                                                    |
- | uiConfig             | {}               | Configuration options for [Swagger UI](https://github.com/swagger-api/swagger-ui/blob/master/docs/usage/configuration.md). Must be literal values, see [#5710](https://github.com/swagger-api/swagger-ui/issues/5710).                                                                                                            |
+ | uiConfig             | {}               | Configuration options for [Swagger UI](https://github.com/swagger-api/swagger-ui/blob/master/docs/usage/configuration.md). Functions should be self contained, as variables outside the functions will not work. You can access the swagger-ui element with `ui` as it is assigned to `window.ui`.                                                                                                   |
```
Comment (Eomm) [inline README.md:111]: Could we add a reference to a test example?

I'm afraid this will become an endless request of examples

Comment (Uzlopak) [reply README.md:111]: uiConfig with onComplete where we do an alert?

Something like this?

```js
'use strict'

const fastify = require('fastify')({ logger: true })

fastify.register(require('@fastify/swagger'), {
  mode: 'static',
  specification: {
    path: './examples/example-static-specification.json'
  }
})

fastify.register(require('../index'), {
  uiConfig: {
    onComplete: () => { alert('test') }
  }
})

fastify.listen({ port: 3000 }, (err) => {
  if (err) throw err
})

```

**Comment by Uzlopak** [comment]
@mcollina 
Should the serializer be a separate npm package?

File: README.md (hunk 5 above)
Comment (Eomm) [reply README.md:111]: Yes - and/or 

```
initOAuth: {
      scopes: ['openid', 'profile', 'email', 'offline_access']
    }
```

if the test inject a request to the new endpoint we can just check the the data are rawly serialized and the flow is complete and working

Comment (Uzlopak) [reply README.md:111]: Yeah. I will probably also add some section into the readme.

**Comment by mcollina** [comment]
No I don't think it's needed at this time.

**Comment by Uzlopak** [comment]
@Eomm PTAL

**Review by mcollina — APPROVED** [review]
lgtm

**Pull request merged**